File: ShadowFrameworkLite20/src/shadow/pipeline/expression/SFBasicExpressionGenerator.java
package shadow.pipeline.expression;


public class SFBasicExpressionGenerator implements SFIExpressionGenerator {

	@Override
	public SFExpressionOperator getOperator(String operatorSymbol){
		if(operatorSymbol.equalsIgnoreCase("+")){
			return new SFExpressionSum();
		}
		if(operatorSymbol.equalsIgnoreCase("*")){
			return new SFExpressionMult();
		}
		if(operatorSymbol.equalsIgnoreCase("/")){
			return new SFExpressionDivide();
		}
		if(operatorSymbol.equalsIgnoreCase("-")){
			return new SFExpressionMinus();
		}
		if(operatorSymbol.equalsIgnoreCase(",")){
			return new SFExpressionVector();
		}
		return new SFExpressionSum();
	}
	
	@Override
	public SFExpressionElement getExpressionElement(String value, short type){
		return new SFExpressionVariable(value,type);
	}
	
	@Override
	public SFExpressionTypeWrapper getWrapper(short type){
		return new SFExpressionTypeWrapper(type);
	}
	
	@Override
	public SFExpressionOperator getFunction(String functionSymbol) {
		return new SFFunctionOperator(functionSymbol);
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/pipeline/expression/SFBasicExpressionGenerator.js
//Java to JS on 21/03/2012

function SFBasicExpressionGenerator(){
}

SFBasicExpressionGenerator.prototype["getOperator"]=function(operatorSymbol){
			if(operatorSymbol=="+"){
				return new SFExpressionSum();
			}
			if(operatorSymbol=="*"){
				return new SFExpressionMult();
			}
			if(operatorSymbol=="/"){
				return new SFExpressionDivide();
			}
			if(operatorSymbol=="-"){
				return new SFExpressionMinus();
			}
			if(operatorSymbol==","){
				return new SFExpressionVector();
			}
			return new SFExpressionSum();
		};

SFBasicExpressionGenerator.prototype["getExpressionElement"]=function(value,type){
			return new SFExpressionVariable(value,type);
		};		
		
SFBasicExpressionGenerator.prototype["getWrapper"]=function(type){
			return new SFExpressionTypeWrapper(type);
		};	
		
SFBasicExpressionGenerator.prototype["getFunction"]=function(functionSymbol){
			return new SFFunctionOperator(functionSymbol);
		};	
		
